### ISSUE TITLE
Imitation Learning ML training needs to be added somewhere

### ISSUE THREAD

**Issue by StefanCaldararu** [uwsbel/autonomy-research-testbed#39]
@harryzhang1018 and I have a seperate repo where we trained the Keras files for the HIL imitation learning models. This should live somewhere related to this repo, but:

- it contains lots of data stored in csv files. where should this be stored?
- it is not directly related to the a-stack, more of a submodule.

here is a link to the current repo:
https://github.com/harryzhang1018/ML_Control 

@AaronYoung5 any suggestions?

**Comment by AaronYoung5** [comment]
Can it live [here](https://github.com/uwsbel/autonomy-research-testbed/tree/master/workspace/src/external)?

And does it need to be a different repo? Is it used for other applications? It could also go at the top level of this repo in a subfolder called utilties or something

**Comment by AaronYoung5** [comment]
@StefanCaldararu @harryzhang1018 still relevant? What's the thoughts here? Ig i wasn't part of any conversations relating to this

**Comment by StefanCaldararu** [comment]
> @StefanCaldararu @harryzhang1018 still relevant? What's the thoughts here? Ig i wasn't part of any conversations relating to this

Yes I think it is still relevant. IMO should be somewhat separated from this repo, but doesn't have to be completely separate. Utilities might be a good solution. Sorry I haven't had much time for this cleanup related stuff lately, maybe @harryzhang1018 can help out more? :)

**Comment by harryzhang1018** [comment]
> > @StefanCaldararu @harryzhang1018 still relevant? What's the thoughts here? Ig i wasn't part of any conversations relating to this
> 
> Yes I think it is still relevant. IMO should be somewhat separated from this repo, but doesn't have to be completely separate. Utilities might be a good solution. Sorry I haven't had much time for this cleanup related stuff lately, maybe @harryzhang1018 can help out more? :)

Yes, I can jump in for this. Currently the Imitation Learning (IL) repo is still in a mess. Definitely need to do clean up. 

Another thoughts: for icra, we did lots of outdoor testing, which data has been stored currently in this ML_Control repo. Any thoughts where should we put these data files?

**Comment by StefanCaldararu** [comment]
> > > @StefanCaldararu @harryzhang1018 still relevant? What's the thoughts here? Ig i wasn't part of any conversations relating to this
> > 
> > 
> > Yes I think it is still relevant. IMO should be somewhat separated from this repo, but doesn't have to be completely separate. Utilities might be a good solution. Sorry I haven't had much time for this cleanup related stuff lately, maybe @harryzhang1018 can help out more? :)
> 
> Yes, I can jump in for this. Currently the Imitation Learning (IL) repo is still in a mess. Definitely need to do clean up.
> 
> Another thoughts: for icra, we did lots of outdoor testing, which data has been stored currently in this ML_Control repo. Any thoughts where should we put these data files?

Some sort of webpage would be nice to be able to display past results relating to ART. bunch of videos, datasets, and maybe even images of the containers so we can go back to past result states easily? IDK that's probably a lot of work but would be cool...

**Comment by harryzhang1018** [comment]
> > > > @StefanCaldararu @harryzhang1018 still relevant? What's the thoughts here? Ig i wasn't part of any conversations relating to this
> > > 
> > > 
> > > Yes I think it is still relevant. IMO should be somewhat separated from this repo, but doesn't have to be completely separate. Utilities might be a good solution. Sorry I haven't had much time for this cleanup related stuff lately, maybe @harryzhang1018 can help out more? :)
> > 
> > 
> > Yes, I can jump in for this. Currently the Imitation Learning (IL) repo is still in a mess. Definitely need to do clean up.
> > Another thoughts: for icra, we did lots of outdoor testing, which data has been stored currently in this ML_Control repo. Any thoughts where should we put these data files?
> 
> Some sort of webpage would be nice to be able to display past results relating to ART. bunch of videos, datasets, and maybe even images of the containers so we can go back to past result states easily? IDK that's probably a lot of work but would be cool...

like this idea. Videos, datasets are easy. But idk about images, cuz we kinda lost track on this for long time and did twist inside container. But, I think it's important to keep in mind for this, for example, after outdoor convoy project, we should work with Thomas to create a image specific for this job.

**Comment by AaronYoung5** [comment]
Shouldn't that go in public-metadata or chrono-reproducibility or whatever that repo is called? I agree docker is the way to go for reproducibility, but I don't think that would be a good place to store all data for projects. Do we need to keep around all the data? Will we ever reuse?

**Comment by harryzhang1018** [comment]
> Shouldn't that go in public-metadata or chrono-reproducibility or whatever that repo is called? I agree docker is the way to go for reproducibility, but I don't think that would be a good place to store all data for projects. Do we need to keep around all the data? Will we ever reuse?

Yea, I think for pic and results, maybe create separate projects on ART website might be helpful and cool.

**Comment by StefanCaldararu** [comment]
> Shouldn't that go in public-metadata or chrono-reproducibility or whatever that repo is called? I agree docker is the way to go for reproducibility, but I don't think that would be a good place to store all data for projects. Do we need to keep around all the data? Will we ever reuse?

agreed. Not all data needs to stay, but we have a dataset that we trained on we should DEFINITELY keep that somewhere and not lose it

**Comment by AaronYoung5** [comment]
good to bring this up with Dan/others to see thoughts. I can also see making a projects page [under this url](https://projects.sbel.org/) (like https://projects.sbel.org/autonomy-toolkit) but with public-metadata-like stuff (like a webpage for each paper we write with embedded images, links to datasets, docker image pull links, code, etc.). 

Either way, github isn't probably the best place to store datasets. We an always link to it in readmes, but probs wanna put in box or drive and link.
